### PR TITLE
AzureMonitor: Ensure original query properties are included unchanged

### LINKED ISSUE #49641: AzureMonitor: Query not created correctly when accompanied by Expression
First reported in [grafana/support-escalations#2709](https://github.com/grafana/support-escalations/issues/2709).

Azure monitor queries that are referenced in an expression query fail as the datasource is not set (the same would also be true for a log query or resource graph query).

How to reproduce:

- Create a dashboard with an Azure monitor query
- Add an expression query
- See the query will fail with an internal server error

### PR DESCRIPTION
**Pull request by aangelisc**
Fixes #49641

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts [hunk 1]
```diff
@@ -134,6 +134,7 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
     const query = templateSrv.replace(item.query, scopedVars, interpolateVariable);
 
     return {
+      ...target,
       refId: target.refId,
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts:138]: you can know remove this line (the same for others)

**Review by andresmgot — COMMENTED** [review]
Can you add tests for this? Thanks!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts (hunk 1 above)
Comment (aangelisc) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts:138]: I may delete this PR in favour of #49543

Comment (andresmgot) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts:138]: I still think that having both contributes to more robust code :)

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts [hunk 2]
```diff
@@ -369,4 +372,37 @@ describe('AzureLogAnalyticsDatasource', () => {
...
+  describe('When performing interpolateVariablesInQueries for azure_log_analytics', () => {
+    beforeEach(() => {
+      templateSrv.init([]);
+    });
+
+    it('should return a query unchanged if no template variables are provided', () => {
+      const query = createMockQuery();
+      query.queryType = AzureQueryType.LogAnalytics;
+      const templatedQuery = ctx.ds.interpolateVariablesInQueries([query], {});
+      expect(templatedQuery[0]).toEqual(query);
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts:385]: we can add a check here to make sure that query.datasource is defined to be extra safe

**Review by andresmgot — APPROVED** [review]
LGTM!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts (hunk 2 above)
Comment (aangelisc) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.test.ts:385]: This is a good shout 😊

**Pull request merged**